### DISCUSSION TITLE
1010todd's Plugin List

### DISCUSSION THREAD

**Discussion by 1010todd** [endless-sky/endless-sky#7928] [Show and tell]
I thought probably should publish more of my plugin outside of Discord so here I am. This will be a list of some of the plugins I've made for the game so far. Though I do publish some of my plugins to https://endless-sky.github.io/plugins.html most of my plugins are not there either because I was lazy, or feel like the quality is too poor or I just don't feel like clogging the official list with random stuffs so here I'll publish most of my plugins. I will also constantly update this post (if I don't forget).

Updated: 2025-12-07 (Add A Number of Starts)
# Large Projects
These are some larger projects that do (or will) do more than just adding a few ships and a gun.

### Galactic War
**Status**: Alpha, very unfinished
This plugin add an entire new galaxy larger than Milky Way with storylines planned and being worked on(none released yet as of 2025-11-05). Qualities varies over the area (partially due to the plugin being so old) and balance can be very questionable in later "tier". Currently released as alpha version for general testing, many things will be changed later on but I'll try to maintain compatibility with older save. Probably one of the oldest plugin still around(or rather revived) and updated.

Requirement: ES 0.10.13+ (I might forgot to update this, check plugin.txt for updated requirement)
https://github.com/1010todd/Galactic-War

Hi-DPI (install in addition to the main plugin above if you want hi-res graphics)
https://github.com/1010todd/Galactic-War-hi-dpi

### Mega Freight

![mgf bluewhale-a](https://user-images.githubusercontent.com/19187937/212466872-aabcfd79-1008-4d4a-aaed-63adc9d0cb75.png)

**Status**: Release, but breaks vanilla balance
As the name implies this plugin add a bunch of freighters, specifically very big heavy freighters but also a few smaller ships including starter-level ship. In addition to that this plugin also add a few extra thing for mining (jobs, weapon, ship, etc.) and a few warships (and very big carriers). 
NPC fleets will be using the ships to add a bit more variety to human space (actually the main reason I made this).
This plugin will break vanilla balance quite a bit. Storylines may also be more difficult with the new ships.

Requirement: ES 0.10.15+
https://github.com/1010todd/Mega-Freight

### Beyond the Sky

![quarg psuchawrl thumb](https://user-images.githubusercontent.com/19187937/212466912-ceac2c31-219b-43e1-b78a-06c5913de558.png)

**Status**: Alpha, potential major changes
Add a couple new ships for higher tier races (Quarg and Pug). Most deployed only after you attack them enough in retaliation. Somewhat balanced on existing content so not really balanced. If you really pissed them off Quarg might also hunt you down if you're close to their system.

Requirement: ES 0.9.17+/Continuous 
https://github.com/1010todd/Beyond-the-Sky

### Endless Endless Sky
**Status**: Alpha, prototype/proof-of-concept
A plugin with Python script to procedurally generate entire galaxy along with the content in it. Due to engine limitation you have to run the Python script manually before launching the game. Pre-generated data is also provided. Many things are still really whack, ship outfitting is awful and half-broken but at least there a new galaxy to explore and half of the stuffs works, hopefully.

Credit Nucleartaxi for galaxy generation script and baseline for ships and outfit scripts.

Requirement: ES 0.9.17+/Continuous, Python 3.9+ (to run the script), Pillow 8.4.0+ (For ship sprite generation)
https://github.com/1010todd/Endless-Endless-Sky

### Ultaka Invasion

![ultaka terminus icon](https://user-images.githubusercontent.com/19187937/212467771-68ce852c-a6ad-409b-bf35-22222221541d.png)

**Status**: Alpha, unfinished
Add new alien that will invade a few areas, need to be triggered with Omis dev/cheat plugin because it will mess up with many things. The invaders are quite OP and unbalanced so you might want a several hundred spare ships to throw into the meat grinder. Comes with mechanic where you can recapture planet from them (by hailing planet and tribute them). Very likely will break storylines. Some cross-plugin compatibility where invader will attack nearly every other plugin's government that I'm aware of and will trigger Quarg to deploy new ships if Beyond the Sky is installed.
A recreation of one of my first plugin (that I never really released).

Requirement: ES 0.9.15+, Galactic War(for asset), Omnis(to trigger invasion)
https://github.com/1010todd/Ultaka-Invasion

### Colony Mechanic
**Status**: Preview/Proof-of-concept
Add "mechanic" for colonizing uninhabited planets (currently just planet Hope for testing). Start by getting Colony Prospector from advanced outfitters. Currently have:
* Various buildings from mines and factories to outfitters and shipyards each with benefits and can buff another related industry. 
* Pirate raids and Meteor landing which can be countered by defenses.

WARNING: Heavily underconstruction, things can change a lot and compatibility can break between versions. Plugin released for testing purpose only, not recommended for general gameplay.

Requirement: ES 0.9.17/Continuous (after 20th December 2022)
[Colony Mechanics.zip](https://github.com/endless-sky/endless-sky/files/10319455/Colony.Mechanics.zip)

# Smaller Projects
Small and miscellaneous plugin that add a few things. 

### Mining Ships

![minsh beetle](https://github.com/endless-sky/endless-sky/assets/19187937/61804672-5554-4976-8cd4-ff48a8cfc026)

**Status**: Release
Add 5 new ships designed for asteroid mining. Might be OP because a few people complained about it back when I released it right after minable asteroids were a thing. Updated with new sprites and some rebalance.

Requirement: ES 0.10.13+
[Mining_ships.zip](https://github.com/user-attachments/files/23363369/Mining_ships.zip)

### Ironclad

![image](https://github.com/endless-sky/endless-sky/assets/19187937/77ed4f0a-ea5d-4873-b0df-282a92611eb4)

**Status**: Functional but Abandoned
Add a human faction that uses a few shield-less ships and weapons in the same style as WW II warships. An update of one of my old plugin.

Requirement: ES 0.9.15+
[Ironclad.zip](https://github.com/endless-sky/endless-sky/files/10296394/Ironclad.zip)

### Prototype D

![prototype d thumb](https://user-images.githubusercontent.com/19187937/212468266-cd723a48-e55b-45d4-a5b7-a9b3a7ba0c4c.png)

**Status**: Release
Add a ship sold at Tarazed after Kestrel was unlocked.

Requirement: ES 0.9.14+
[prototype D.zip](https://github.com/endless-sky/endless-sky/files/10296835/prototype.D.zip)

### Fancy Weapon
**Status**: Example plugin/Proof-of-concept
Not for normal gameplay. This plugin have a few weapons with fancy firing pattern using complicated submunition chain that can be used as example/reference if you want to make weird weapons.

Requirement: ES 0.9.15+
[Fancy Weapons.zip](https://github.com/endless-sky/endless-sky/files/10296350/Fancy.Weapons.zip)

### Nuke Turret

![icon](https://user-images.githubusercontent.com/19187937/212468204-6c4e88ba-6715-40c8-a87b-f935bcc9d40a.png)

**Status**: Release
Add nuclear launcher turret. Unlocked at the same time Nuclear Missile is unlocked.

Requirement: ES 0.9.15+
[Nuke turret.zip](https://github.com/endless-sky/endless-sky/files/10296374/Nuke.turret.zip)

### Crowded Sky
**Status**: Release
Increase spawnrate of fleets by 10x because someone asked for it. Just for fun. (Can get overridden by events.)

Requirement: ES 0.10.0
[Crowded Sky 0.10.0.zip](https://github.com/endless-sky/endless-sky/files/10817834/Crowded.Sky.0.10.0.zip)

### Punt Gun

![birdshot](https://user-images.githubusercontent.com/19187937/212467852-94fb6bab-83f7-4cb0-926f-bb29768e6d58.PNG)

**Status**: Release
Add high recoil shotgun weapon based on Proton Gun. Sold by Kraz Cybernetics. NPCs also use them.

Requirement: ES 0.9.15+
[Punt Gun.zip](https://github.com/endless-sky/endless-sky/files/10296463/Punt.Gun.zip)

### Extra Large Jobs
**Status**: Release
Add several jobs with larger cargo and bunk space requirement for plugins that add larger ships. Texts might not make the most sense.

Requirement: ES 0.9.15+
[Extra Large Jobs.zip](https://github.com/endless-sky/endless-sky/files/10296496/Extra.Large.Jobs.zip)

### Mitsu Industry
**Status**: Release
Add a shipyard that sell a few model of very small car-like ships.

Requirement: ES 0.9.15+
[Mitsu Industry.zip](https://github.com/endless-sky/endless-sky/files/10296547/Mitsu.Industry.zip)

### 50cal
**Status**: Release
Add small M2 browning as outfit to fit in the tiniest gap left on your ship. Also added M2 and M82 as H2H.

Requirement: ES 0.9.15+
[50 cal.zip](https://github.com/endless-sky/endless-sky/files/10296573/50.cal.zip)

### Fighter Factory

![ff fighter assembler](https://github.com/endless-sky/endless-sky/assets/19187937/cd1a672b-9aad-4791-bf6a-7faab10f2270)

**Status**: Beta
Add outfit that can produce fighters to replace lost ones every landing. (Currently support Human fighters/drone and Remnant fighters/drone)
To use after install the Fighter Assembler on the __flagship__ depart and land again then check the job board and accept "!F: Record Fighter" job to make it remember the current number of fighter and drones, if they fall below the remembered number the Assembler will make a new fighter/drone to replace it based on blueprint installed. Also support one custom fighter blueprint if you're willing to edit the file (Edit ffcustomfighter.txt and replace the model and outfit with your own thing will keeping "FiFc: Custom Fighter 1" after the model name.).

Requirement: ES 0.9.15+
https://github.com/1010todd/Fighter-Factory

### Drop Loot
**Status**: Beta
Allow installed outfits to be dropped with chances based on mass of the outfit. Outfits with existing chance will not be affected. There's no other filter so it can be quite wild as it can drop supposedly unobtainable outfits.

Requirement ES 0.10.16+/Continuous ~~(commit  03d1414 / after 29th December)~~
[Drop_Loot-0.10.16a.zip](https://github.com/user-attachments/files/23582149/Drop_Loot-0.10.16a.zip)

### Large Asteroid

![hugeasteroid](https://user-images.githubusercontent.com/19187937/212467048-7ba02035-ba45-4ad7-adac-bdddcf2b3f10.PNG)

**Status**: Beta
Add bigger asteroid as mineable randomly to systems. Asteroids are often big enough for interceptors to even hide inside and in general just a more useful size for actually dodging something. Since mineable asteroid orbits in belt it also provide some terrain considerations.

Now also support adding to Galactic War systems.

Requirement: ES 0.10.16+
https://mega.nz/file/FVJ1RJzJ#rKAHOA0wydEB_1oQhRJ_9OQBY0WTTDcUNM78-kxfG6s

### Prong

![prong thumb](https://user-images.githubusercontent.com/19187937/212466785-2c76354d-addf-48b6-8aa7-319f0c798902.png)

**Status**: Beta
Add Prong interceptor, a Barb with another gun pod tacked on.

Requirement: ES 0.9.15+
[Prong.zip](https://github.com/endless-sky/endless-sky/files/10411007/Prong.zip)

### Improvisatory 

![imv gunbarge thumb](https://user-images.githubusercontent.com/19187937/212466776-00e987a8-36ac-477c-9a4c-f05335c7444e.png)

**Status**: Beta
Add a few ships that are a modification of existing ships.

Requirement: ES 0.9.15+
[Improvisatory.zip](https://github.com/endless-sky/endless-sky/files/11325446/Improvisatory.zip)

### Scrapper Extras

![scrapper slugger thumb](https://user-images.githubusercontent.com/19187937/216760069-57c0351c-727b-402c-8c9c-5b05f8fe5e64.png)

**Status**: Release
Add a few more variants of Scrappers that didn't made it into vanilla and a few special modified variant with unique weapons. Note: due to amount of variants added Scrappers will become a lot more common.

Requirement: ES 0.10.0+/Continuous
[Scrapper Extras.zip](https://github.com/endless-sky/endless-sky/files/14512651/Scrapper.Extras.zip)

### Tremor Gunship

![ags tremor thumb](https://user-images.githubusercontent.com/19187937/216760152-bb414b12-23bc-4e17-a764-73532f0a3ce9.png)

**Status**: Beta
Add a quite OP Light Warship with unique weapon that could theoretically handle most ship on its own with some piloting skill.

Requirement: ES 0.10.0+/Continuous
[Tremor Gunship.zip](https://github.com/endless-sky/endless-sky/files/10608314/Tremor.Gunship.zip)

### Remove Invincible Asteroid

![icon](https://user-images.githubusercontent.com/19187937/224357178-7b2ae5a3-a651-4890-a560-fcbff4af5945.png)

**Status**: Release
Remove the annoying invincible, infinite asteroid field. 

Now also support systems from Galactic War.

Requirement: ES 0.10.16+
[Remove_Invincible_Asteroid-0-10-16.zip](https://github.com/user-attachments/files/23579082/Remove_Invincible_Asteroid-0-10-16.zip)

### Green Mereti

![gmrt model512 greenhouse thumb](https://user-images.githubusercontent.com/19187937/225725860-555ce7f4-b4fa-4dbf-a399-5f0587d8c237.png)

**Status**: Beta
Add new Mereti ship deployed a while after Wanderer Mind merged with Kor Mereti and the Sestor is eliminated. The new ships have more terraforming-focused purpose.

Requirement: ES 0.10.0+/Continuous
[GreenMereti.zip](https://github.com/endless-sky/endless-sky/files/10994875/GreenMereti.zip)

### Heavy Blaster

![heavy blaster](https://user-images.githubusercontent.com/19187937/225726291-5a80124f-f782-411b-8f93-d3f12d21331b.png)

**Status**: Release
Add heavy blaster as generic midway between Twin Blasters and Proton Gun/Plasma Cannon but more largely available. NPCs will also use it.

Now with turret!

Requirement: ES 0.10.0+/Continuous
[Heavy Blaster.zip](https://github.com/endless-sky/endless-sky/files/14560119/Heavy.Blaster.zip)

### More Merchant Name

![More Merchant Name icon](https://user-images.githubusercontent.com/19187937/233645642-2c493a8a-4e8a-4919-9a1d-9a6bb9162b9e.png)

**Status**: Beta
Add a bunch more possible names for merchant based on/inspired by real life ship names and combination along with adding several non-English names. (Now also include new bounty hunter and pirate names)

Requirement: ES 0.9.16+/Continuous
[More Merchant Name.zip](https://github.com/endless-sky/endless-sky/files/14388562/More.Merchant.Name.zip)

### Dregs

![dregs thumb](https://user-images.githubusercontent.com/19187937/233639657-93555102-fa4d-4aff-8f39-ddbdf2da612d.png)

**Status**: Release
Add a fighter called Dregs across a few planets.

Requirement: ES 0.9.16+/Continuous
[Dregs.zip](https://github.com/endless-sky/endless-sky/files/11295799/Dregs.zip)

### MGF-PFlame Compatibility

**Status**: Alpha
Bridge for Mega Freight and Prometheus Flame plugin currently adding Mega Freight fleets to new systems as appropriate and includes Prometheus Flame ships in several Mega Freight fleets.

Requirement: ES 0.10.0+/Continuous, Mega Freight plugin, Prometheus Flame plugin
[MGF-PFlame Compat.zip](https://github.com/endless-sky/endless-sky/files/11325468/MGF-PFlame.Compat.zip)

### Starship Architect

![icon](https://github.com/endless-sky/endless-sky/assets/19187937/3f881757-2515-44bd-9796-906388364968)

**Status**: Beta
Add (currently two) ship with calculated mass and internal volume as well as detailed interior and structure design to get better guess of how large and heavy ships should be. The ship is accessible via Omnis plugin. Also more general semi-ES-balanced mass version used by merchants and accessible in normal gameplay.

Requirement: ES 0.9.16+/Continuous
[Starship Architect.zip](https://github.com/endless-sky/endless-sky/files/14529542/Starship.Architect.zip)

### Nuke Gat

![nuke gat](https://github.com/endless-sky/endless-sky/assets/19187937/e59d8aec-6784-47d7-a59c-043e3e84bc7a)

**Status**: Beta
Rapid firing nuke launcher.

Requirement: ES 0.9.16+/Continuous
[Nuke Gat.zip](https://github.com/endless-sky/endless-sky/files/12170229/Nuke.Gat.zip)

### VeryHeavy Dreadnought

![space dreadnought thumb](https://github.com/endless-sky/endless-sky/assets/19187937/d4c5186c-285f-400f-af52-cafe9f9af990)

**Status**: Beta
Someone asked for a Dreadnought.

Requirement: ES 0.10.7+/Continuous (Turret Arc):
[VeryHeavy Dreadnought.zip](https://github.com/user-attachments/files/16059704/VeryHeavy.Dreadnought.zip)

Requirement: ES 0.10.2+/Continuous (No Turret Arc):
[VeryHeavy Dreadnought.zip](https://github.com/endless-sky/endless-sky/files/12170281/VeryHeavy.Dreadnought.zip)

### Mega Freight: Weapon Pack

![mgf ccs j45 shaixuan](https://github.com/endless-sky/endless-sky/assets/19187937/388d01a3-0819-4899-93eb-7abf117739f6)

**Status**: Beta
Add more weapons and ships to use them because I'm bored of hearing Proton Turrets all the time. May cause some error without Mega Freight as it adds outfits to Mega Freight ships as well.

Requirement: ES 0.10.16+/Continuous, Mega Freight (Optional)
[MGF_Weapon_Pack-0.3.0.zip](https://github.com/user-attachments/files/23836213/MGF_Weapon_Pack-0.3.0.zip)

### A Number of Starts

<img width="180" height="170" alt="argosy" src="https://github.com/user-attachments/assets/0eb15d39-5792-4a32-9a4c-a01771a46f2d" />

**Status**: Beta
Add a few new starts each with some uniqueness like hardcore Scrapper start, start with a trillion in debt, start that will send bounty hunters after you often, etc.

Requirement: ES 0.10.16+/Continuous
[A_Number_of_Starts-0.1.0.zip](https://github.com/user-attachments/files/23998117/A_Number_of_Starts-0.1.0.zip)

**Comment by Zireael07** [comment]
I like the idea behind Endless Endless Sky, will be giving the code a look!

**Comment by bene-dictator** [comment]
I've been playing around with and changing the stats of your Overdriven Energy Blaster (from Scrapper Extras) in one of my branches. It's actually quite fun to use on interceptors and I wish it were in the main branch, only that we have so many energy blaster variants now that it's unlikely to get in.

**Comment by Zitchas** [comment]
Nice work!

Personally, I'd say all of them deserve to be in the plug-in repository. Any problems about "clogging the list" are problems with the plug-in list, not the plug-in. It's not supposed to be a curated list of "top" or "official" plug-ins, it's supposed to be a list of *all* plug-ins, from which the player can pick and choose what they want. Perhaps we need better ways of tagging and flitering plug-ins to help people find what they want.

Anyway, lots of interesting plug-ins here. Larger Asteroids would solve a complaint I've often had that asteroids aren't big enough to be useful for anything other than imposing a random fail chance on weaponry. I'd love to have half as many asteroids, but have asignificant fraction of them be big enough to actually use as cover.

**Reply by 1010todd** [reply]
I suppose having way to tag very in-progress plugin would help as part of the reason I didn't put some from here on the list is because it's still too far in development and not really ready for gameplay. But mostly it's the laziness on my part to create a new repo and PR the plugin to the list. I could probably try to add some smaller one to the list over time.

**Reply by samrocketman** [reply]
It can auto update by Git commit instead of tags if you prefer.  For example, https://github.com/endless-sky/endless-sky-plugins/pull/150/files

**Reply by 1010todd** [reply]
"Tag" here meant flag/label/whatever word to categorize something (in this case, a plugin) not Github tag and/or release thing. I already used commit autoupdate for my other plugins that are on the list. 

The main problem is the entire process of making that exact file you linked, putting it in a git branch, and making a pull request for it to get into the list is way too complicated and time consuming than just drag and drop a zip file into a post and be done with it.

**Comment by lumbar527** [comment]
The Large Asteroids plugin is fun, though it can give you a little too much money. I've found a system with frequent _huge gold_ asteroids (not telling you where). With my Behemoth I bought nearby, I got several million credits for a full cargo hold.

**Comment by OcelotWalrus** [comment]
All your plugin have so good ideas and that give me so many other ideas & concepts, congrats for your great mods!

**Comment by ghost** [comment]
If you're interested, could I ask you to make a turreted version of the Heavy Blaster mounting two guns? You do such nice graphics for everything, and it's the exact kind of thing I've always felt the game needed, just a normal medium-heavy turret without any fancy damage types or anything.

**Reply by 1010todd** [reply]
Added Heavy Blaster turret.

**Reply by ghost** [reply]
Thank you! Looks awesome.

These aren't issues for me, but there are a few oversights:
- Hardpoint lacks a high-DPI graphic.
- It's still in the Guns category and fills a gun port rather than a turret slot (also lacks crew requirement).
- "Hardpoint sprite" line says hardpoint instead of "hardpoint sprite" (including quotes) so the graphic doesn't appear on ships.
- "Hardpoint offset" line is a child of "hardpoint sprite" line; shouldn't be.

**Reply by 1010todd** [reply]
Hardpoints are tiny I think 2x doesn't really make much difference if at all. 

Anyway, thanks for the report. Update with the fix.

**Comment by TheGiraffe3** [comment]
Are you planning on PRing More.Merchant.Name anytime soon? I think it would be nice to have more variety.
~~get it in for 0.10.9~~
